linera-views/tests/operations_tests.rs

```diff
@@ -205,103 +208,180 @@ async fn test_readings_memory_specific() {
+        batch.put_key_value_bytes(key2, vec![]);
+        batch.put_key_value_bytes(key3, vec![]);
+        //
+        batch.delete_key(key1);
+        let key_prefix = get_key(key_prefix, vec![12]);
+        batch.delete_key_prefix(key_prefix);
+    }
+    batch
+}
+
+fn realize_batch(batch: &Batch) -> BTreeMap<Vec<u8>, Vec<u8>> {
+    let mut kv_state = BTreeMap::new();
+    for operation in &batch.operations {
+        match operation {
+            WriteOperation::Put { key, value } => {
+                kv_state.insert(key.to_vec(), value.to_vec());
+            }
+            WriteOperation::Delete { key } => {
+                kv_state.remove(key);
+            }
+            WriteOperation::DeletePrefix { key_prefix } => {
                 let key_list = kv_state
-                    .range(get_interval(delete_key_prefix.clone()))
+                    .range(get_interval(key_prefix.clone()))
                     .map(|x| x.0.to_vec())
                     .collect::<Vec<_>>();
                 for key in key_list {
                     kv_state.remove(&key);
                 }
             }
         }
-        key_value_store.write_batch(batch, &[]).await.unwrap();
-        // Checking the consistency
-        let mut key_values = BTreeMap::new();
-        for key_value in key_value_store
-            .find_key_values_by_prefix(&key_prefix)
-            .await
-            .unwrap()
-            .iterator()
-        {
-            let key_value = key_value.unwrap();
-            let mut key = key_prefix.clone();
-            key.extend(key_value.0);
-            key_values.insert(key, key_value.1.to_vec());
-        }
-        assert_eq!(key_values, kv_state);
+    }
+    kv_state
+}
+
+#[cfg(test)]
+async fn run_test_batch<OP: KeyValueStoreClient + Sync>(
+    key_value_operation: &OP,
+    key_prefix: Vec<u8>,
+    batch: Batch,
+) {
+    let kv_state = realize_batch(&batch);
+    key_value_operation.write_batch(batch, &[]).await.unwrap();
+    // Checking the consistency
+    let mut key_values = BTreeMap::new();
+    for key_value in key_value_operation
+        .find_key_values_by_prefix(&key_prefix)
+        .await
+        .unwrap()
+        .iterator()
+    {
+        let key_value = key_value.unwrap();
+        let mut key = key_prefix.clone();
+        key.extend(key_value.0);
+        key_values.insert(key, key_value.1.to_vec());
+    }
+    assert_eq!(key_values, kv_state);
+}
+
+#[cfg(test)]
+async fn run_writings<OP: KeyValueStoreClient + Sync>(key_value_operation: &OP) {
+    let mut rng = rand::rngs::StdRng::seed_from_u64(2);
+    let n_oper = 100;
+    let size_batch = 50;
+    // key space has size 4^4 = 256 so we necessarily encounter collisions
+    // because the number of generated keys is about size_batch * n_oper = 800 > 256.
+    let option = 1;
+    for _ in 0..n_oper {
+        let key_prefix = get_random_key_prefix();
+        let batch = generate_random_batch(&mut rng, &key_prefix, size_batch, option);
```

Comment:
Don't we also want to test with random batches with option 2, i.e. renormalized?

Reply:
That is what I was initially thinking.
However one restriction I needed was to make sure that the operations on the keys are independent. So, I eliminated that option.